src/redismodule.h

```diff
@@ -842,6 +842,7 @@ typedef void (*RedisModuleInfoFunc)(RedisModuleInfoCtx *ctx, int for_crash_repor
 typedef void (*RedisModuleDefragFunc)(RedisModuleDefragCtx *ctx);
 typedef int (*RedisModuleDefragFunc2)(RedisModuleDefragCtx *ctx);
 typedef void (*RedisModuleUserChangedFunc) (uint64_t client_id, void *privdata);
+typedef void *(*RedisModuleDefragDictValueCallback)(void *data, unsigned char *key, size_t keylen);
```

Comment:
I'm not sure that we need to pass the `key` and `keylen` as parameters to the callback. But there's no harm in them existing for other users.

However I am fairly certain that we do need to add a `RedisModuleDefragCtx *ctx` argument. Without which, it is impossible to call other defrag functions from within the CB, eg `RedisModule_DefragAlloc`

Reply:
>However I am fairly certain that we do need to add a RedisModuleDefragCtx *ctx argument. Without which, it is impossible to call other defrag functions from within the CB, eg RedisModule_DefragAlloc

sure, it's reasonable.